arangod/Graph/Providers/ClusterProvider.cpp

```diff
@@ -262,15 +264,31 @@ void ClusterProvider<StepImpl>::destroyEngines() {
 }
 
 template<class StepImpl>
-Result ClusterProvider<StepImpl>::fetchEdgesFromEngines(
-    VertexType const& vertex) {
+Result ClusterProvider<StepImpl>::fetchEdgesFromEngines(Step* step) {
+  TRI_ASSERT(step != nullptr);
+
   auto const* engines = _opts.engines();
-  // TODO Assert that the vertex is not in _vertexConnections after no-loose-end
+  // TODO Assert that the step is not in _vertexConnections after no-loose-end
   // handling todo is done.
   transaction::BuilderLeaser leased(trx());
   leased->openObject(true);
-  leased->add("backward", VPackValue(_opts.isBackward()));
-  leased->add("keys", VPackValue(vertex.toString()));
+  leased->add("backward",
+              VPackValue(_opts.isBackward()));  // [GraphRefactor] ksp only?
+
+  // [GraphRefactor] TODO: Differentiate between algorithms -> traversal vs.
```

Comment:
Why do we need to differentiate here?
As far as I see this code now works for all variants correct?

Reply:
It does, but atm we're sending information that is not required in all cases. With specific cases, we could only send the info which is required.

Reply:
OKay, than keep for now